addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -259,16 +308,34 @@ public OpenApiResults importOpenApiDefinitionV2(
             final String targetUrl,
             final String definitionUrl,
             boolean initViaUi,
-            final Requestor requestor) {
+            final Requestor requestor,
+            int contextId) {
         final List<String> errors = new ArrayList<>();
         SwaggerConverter converter =
                 new SwaggerConverter(targetUrl, definitionUrl, defn, getValueGenerator());
+
+        if (contextId >= 0) {
+            try {
+                List<StructuralNodeModifier> structuralNodeModifiers =
+                        converter.getStructuralNodeModifiers();
+                Context ctx = Model.getSingleton().getSession().getContext(contextId);
+                ctx.setDataDrivenNodes(structuralNodeModifiers);
+            } catch (Exception e) {
+                LOG.error(
+                        "Error setting data driven nodes from spec for contextId: " + contextId, e);
```

Comment:
We should use `{}` for log parameters - there are other instances in this PR as well.